examples/drupal_loadtest.rs

```diff
@@ -99,25 +103,34 @@ fn drupal_loadtest_front_page(client: &mut GooseClient) {
         },
     }
 }
+fn drupal_loadtest_front_page_async<'r>(client: &'r mut GooseClient) -> Pin<Box<dyn Future<Output = ()> + Send + 'r>> {
+    Box::pin(drupal_loadtest_front_page(client))
+}
```

Comment:
This is a bit of a baptism of fire into rust async. This is a problem that none of the async web frameworks have a good answer to (as far as I can see).

I've been screwed by this ergonomics problem repeatedly and not found a good answer. I wanted to write GooseTask::new() as a generic function that could accept `drupal_loadtest_front_page` directly, but I couldn't manage it.

This wrapper function seems to be in the privileged position that it is only generic over one thing (the lifetime 'r), and not also over the returned future. (I have previously managed to be generic over returned future in another project ( https://github.com/gotham-rs/gotham/pull/281/files#diff-7ea3539fe8dc99083b2e3fb06a9bcb13R525 ), but not over both at the same time).

I'm wondering if we could cheat by making a macro called task!(), so you do    
```
    GooseAttack::initialize()
        .register_taskset(GooseTaskSet::new("AnonBrowsingUser")
            .register_task(task!(drupal_loadtest_post_comment)
        )
```

It could then generate a closure that does the same boxing job that this wrapper function does, and pass that into GooseTask::new(). This might side-step the problem in the same way.

I might follow up on a few of your threads and see if I can find a solution that avoids macros (or a satisfying explanation of why it's not possible).

Reply:
My apologies -- I have nothing against using Macros. I was just not eager to build on top of `macro_rules_attribute` as it's a 0.0.1 release and has no other software depending on it. I'm actually very eager to add some custom macros to Goose to simplify the creation of load tests, generally speaking.

Your idea of making a task!() macro would be great.

Reply:
This is now working wonderfully with the `task!()` macro.